packages/vscode-dependencies-validation/src/logger/logger.ts

```diff
@@ -0,0 +1,37 @@
+import { ExtensionContext, window } from "vscode";
+import { IVSCodeExtLogger } from "@vscode-logging/types";
+import { configureLogger, NOOP_LOGGER } from "@vscode-logging/wrapper";
+
+export const LOGGING_LEVEL_CONFIG_PROP = "dependenciesValidation.logging.level";
+export const SOURCE_TRACKING_CONFIG_PROP =
+  "dependenciesValidation.logging.sourceLocationTracking";
+
+let logger: IVSCodeExtLogger = NOOP_LOGGER;
+
+/**
+ * Note the use of a getter function so the value would be lazy resolved on each use.
+ * This enables concise and simple consumption of the Logger throughout our extension.
+ */
+export function getLogger(): IVSCodeExtLogger {
+  return logger;
+}
+
+/* istanbul ignore next - ignoring "legacy" missing coverage to enforce all new code to be 100% */
+export function initLogger(context: ExtensionContext): void {
+  const extensionName = "vscode-dependency-validation"; // If the extension name changes, change this too
```

Comment:
i think `context.extension.id` should give you the name

Reply:
if not make a **test** to match this const value with what is in the pkg.json

Reply:
we already have an output channel with this name.